scala2-library-cc/src/scala/concurrent/impl/Promise.scala

```diff
@@ -69,15 +69,15 @@ private[concurrent] object Promise {
     /**
      * Compresses this chain and returns the currently known root of this chain of Links.
      **/
-    final def promise(owner: DefaultPromise[T]^): DefaultPromise[T]^{cap.rd, this, to, owner} = {
+    final def promise(owner: DefaultPromise[T]^): DefaultPromise[T]^{to, owner} = {
       val c = get()
       compressed(current = c, target = c, owner = owner)
     }
 
     /**
      * The combination of traversing and possibly unlinking of a given `target` DefaultPromise.
      **/
-    @inline @tailrec private[this] final def compressed(current: DefaultPromise[T]^{cap.rd, this}, target: DefaultPromise[T]^, owner: DefaultPromise[T]^): DefaultPromise[T]^{this, current, target, owner} = {
+    @inline @tailrec private[this] final def compressed(current: DefaultPromise[T]^{this, to}, target: DefaultPromise[T]^{cap, this, current}, owner: DefaultPromise[T]^): DefaultPromise[T]^{to, target, owner} = {
```

Comment:
We should look into this more, I tried simplifying this signature but it seems separation checker is not trying to shorten paths